src/inference/src/dev/make_tensor.cpp

```diff
@@ -266,22 +289,15 @@ std::shared_ptr<ITensor> make_tensor(const element::Type element_type, const Sha
  */
 class RoiTensor : public ITensor {
 public:
-    RoiTensor(const std::shared_ptr<ITensor>& owner, const Coordinate& begin, const Coordinate& end) : m_owner{owner} {
+    RoiTensor(const std::shared_ptr<ITensor>& owner, const Coordinate& begin, const Coordinate& end)
+        : m_owner{owner},
```

Comment:
When taking ownership passing on `shared_ptr` by value is preferred.
```suggestion
    RoiTensor(std::shared_ptr<ITensor> owner, const Coordinate& begin, const Coordinate& end)
        : m_owner{std::move(owner)},
```